src/tx/builder/schema.js

```diff
@@ -22,7 +22,8 @@ export const RESPONSE_TTL = { type: 'delta', value: 10 }
 // # CONTRACT
 export const DEPOSIT = 0
 export const AMOUNT = 0
-export const GAS = 25000
+export const GAS_ABOVE_AVERAGE = 25000 // TODO: don't use this
+export const GAS_MAX = 1600000 - 21000
```

Comment:
why do you want to introduce this again? this was the default in the past. we should avoid this being used

Reply:
when I read the docs correctly we don't need to provide this param for a dry-run. according to the docs it will be automatically set to `1_000_000`, see https://github.com/aeternity/aeternity/blob/master/apps/aehttp/priv/oas3.yaml#L580

can you check that? as we are using protected dry-run by default anyway advanced devs should know how to deal with gas estimation anyway.

Reply:
> we don't need to provide this param for a dry-run

> Since DryRunCallReq object do not have a mandatory gas field, if not set a default value of 1000000 is being used instead.

Gas can be omitted only if we pass transaction as an object instead of a tx-encoded string.

https://github.com/aeternity/aeternity/blob/6f0113e00766d0b675aab0327292b9d270a1f052/apps/aehttp/priv/oas3.yaml#L3245-L3266

We didn't use this feature before, and for me looks more robust to pass as an encoded string because we also checking that the string is encoded correctly.

Reply:
ok I understand. but why do we use  `GAS_MAX` again? (if it is even a correct value for the max gas)

where do we need this and why?

Reply:
I'm using `GAS_MAX` in dry-run calls (static calls of a contract method) if the user didn't provide any. The gas field can't be omitted because the transaction is tx-encoded at that moment. SDK can't pass any other gas value, because it can be estimated only by this call.

Reply:
ok, got it